pkg/telemetry/workflowsopenshift.go

```diff
@@ -0,0 +1,40 @@
+package telemetry
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kong/kubernetes-telemetry/pkg/provider"
+)
+
+const (
+	// OpenShiftVersionWorkflowName is the name assigned to the OpenShift version detection workflow.
+	OpenShiftVersionWorkflowName = "openshift-version"
+)
+
+// NewOpenShiftVersionWorkflow returns an OpenShift version detection workflow.
+//
+// Example report produced:
+//
+//	{
+//		"openshift_version": "4.13.0",
+//	}
+func NewOpenShiftVersionWorkflow(cl client.Client, namespace, app string) (Workflow, error) {
```

Comment:
While the provider is aptly named I wonder if this workflow couldn't be somehow merged into the existing workflows and reuse what's already available. This way we'll have more context in one place. (in the end everything is serialized into semicolon delimited format but better to think about this now rather than later).

For instance we already have the `IdentifyPlatformWorkflow`, which returns reports with cluster arch, cluster version, cluster semver and cluster provider (the file is called `k8scloudproviderprovider.go` 🤦 but it can be renamed to provide the context of not only cloud but cluster providers in general).

This way e.g. on openshift we'll get a report from that workflow:

```
	{
	  "k8s_arch": "linux/amd64",
	  "k8sv": "v4.13.0",
	  "k8sv_semver": "v4.13.0",
	  "k8s_provider": "OpenShift"
	}
```

WDYT?

Reply:
We can stick it elsewhere fine if we don't mind not being able to disable it (or maybe shove disable logic into the detector, so that it stops running if the initial check fails).

Downstream (when KIC instantiates the telemetry manager) it can choose to toggle checks off by provider, so if we wanted something there to initially check if the namespace exists and disable the provider if not (with the expectation that it's very unlikely to appear after start), it'd need to be separate.

Reply:
I don't see a reason to disable it. If we put it into the identify platform workflow then it's just going to be one more platform to detect and one more set of rules that will run as part of that (other platforms are e.g. detected via labels and/or annotations, this one will use ENVs).

WDYT of that?

Reply:
> It seems silly to continuously ping the API server for something that's never going to change as long as the Pod lives (you'd need to restart the kubelet and everything running on it by extension). That's true for the provider/arch/Kubernetes version also.

> Would it make sense to cache the results of checks that won't change after startup? That'd be better than relying on downstream to handle disabling it.

I agree with that but let's tackle is separately 👍 

> I think we wouldn't stick this in the cloud provider field. This isn't a provider so much as it's a distribution, and you can still run OpenShift on GCP and AWS (though IDK if our detector can tell the underlying provider if you've rolled your own install and are not using EKS or GKE), and I'd expect we're still interested in that.

I can see your point. The only thing I'm trying to avoid is to make this an OpenShift special case because it's not. Your remark about it not being a cloud provider is valid 💯. In such case maybe we introduce a new notion of distribution or distro? Initially we would only detect OpenShift but if need be we'd extend it with k3s, kind and whetever else we deem worthy. WDYT?